### PR TITLE
Classifier transformer

### PR DESCRIPTION
**Pull request by abhishekraok**
Fix dev and test set metrics calculation.
Add transformers classifier.

### CODE REVIEW

File: coconut/main/train_bert_classifier.py [hunk 1]
```diff
@@ -0,0 +1,1019 @@
...
+    pearson_corr = pearsonr(preds, labels)[0]
+    spearman_corr = spearmanr(preds, labels)[0]
+    return {
+        "pearson": pearson_corr,
+        "spearmanr": spearman_corr,
+        "corr": (pearson_corr + spearman_corr) / 2,
+    }
+
+
+def compute_metrics(task_name, preds, labels):
```
Comment (mbetser) [inline coconut/main/train_bert_classifier.py]: This is also not very extensible. Is this code actually coming from another repo?

File: coconut/main/train_classifier.py [hunk 2]
```diff
@@ -118,69 +131,87 @@ def write_predictions_for_test_data(iterator: DataIterator, logger: logging.Logg
...
     parser = argparse.ArgumentParser()
     parser.add_argument("--data_path", help="The path to the training, validation and test data", type=str,
                         required=True)
     parser.add_argument("--output_dir", help="Trained Model will be saved here", type=str, required=True)
+    parser.add_argument("--trained_model_dir", help="Optional, Trained Model will be loaded from here, "
+                                                    "instead of random weights", type=str, default='')
     # Hyper Parameters
     parser.add_argument("--batch_size", default=64, type=int)
     parser.add_argument("--embedding_dimension", default=16, type=int)
     parser.add_argument("--embedding_type", default="word", choices=["word", "elmo", "bert"])
```
Comment (mbetser) [inline coconut/main/train_classifier.py]: for model_type "transformers" the embedding should be always bert?

File: coconut/main/train_classifier.py [hunk 3]
```diff
@@ -118,69 +131,87 @@ def write_predictions_for_test_data(iterator: DataIterator, logger: logging.Logg
...
+    parser.add_argument("--model_type", default="rnn", choices=["rnn", "transformers"])
```
Comment (mbetser) [inline coconut/main/train_classifier.py]: since rns and bert are very different models, maybe it would be better to have 2 separate training scripts. But factorizing the common code as much as possible into a common class?

File: coconut/main/train_bert_classifier.py [hunk 4]
```diff
@@ -0,0 +1,1019 @@
+"""Code from https://github.com/huggingface/pytorch-pretrained-BERT
```
[the remark this answers is not in the document]
Comment (abhishekraok) [reply coconut/main/train_bert_classifier.py]: Yes, good idea. Actually this file was added to this PR by mistak. I will remove it.

File: coconut/main/train_bert_classifier.py [hunk 5]
```diff
@@ -0,0 +1,1019 @@
...
+        # sequence or the second sequence. The embedding vectors for `type=0` and
+        # `type=1` were learned during pre-training and are added to the wordpiece
+        # embedding vector (and position vector). This is not *strictly* necessary
+        # since the [SEP] token unambiguously separates the sequences, but it makes
+        # it easier for the model to learn the concept of sequences.
+        #
+        # For classification tasks, the first vector (corresponding to [CLS]) is
+        # used as as the "sentence vector". Note that this only makes sense because
+        # the entire model is fine-tuned.
+        tokens = ["[CLS]"] + tokens_a + ["[SEP]"]
```
[the remark this answers is not in the document]
Comment (abhishekraok) [reply coconut/main/train_bert_classifier.py]: Previous one had it internally.

File: coconut/main/train_bert_classifier.py (hunk 1 above)
Comment (abhishekraok) [reply coconut/main/train_bert_classifier.py]: yes
